src/HealthChecks.CosmosDb/AzureCosmosDbHealthCheck.cs

```diff
@@ -31,23 +31,21 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
     {
         try
         {
+            if (_options is {DatabaseId: not null, ContainerIds: not null})
+            {
+                foreach (var containerId in _options.ContainerIds)
+                {
+                    var container = _cosmosClient.GetContainer(_options.DatabaseId, containerId);
+                    var query = new QueryDefinition($"SELECT 1 FROM {containerId}");
+                    using var queryStream = container.GetItemQueryStreamIterator(query);
```

Comment:
Don't you need to call .ReadNextAsync() to execute the operation against the container?

Reply:
@idkburkes is correct: This is correct. The query is not executed until you call await ReadNextAsync()